server/src/main/java/org/opensearch/index/translog/listener/CompositeTranslogEventListener.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.listener;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.apache.lucene.store.AlreadyClosedException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The listener that multiplexes other {@link TranslogEventListener}
+ *
+ * @opensearch.internal
+ */
+public final class CompositeTranslogEventListener implements TranslogEventListener {
+
+    private final List<TranslogEventListener> listeners;
+    private final Logger logger = LogManager.getLogger(CompositeTranslogEventListener.class);
+
+    public CompositeTranslogEventListener(Collection<TranslogEventListener> listeners) {
+        for (TranslogEventListener listener : listeners) {
+            if (listener == null) {
+                throw new IllegalArgumentException("listeners must be non-null");
+            }
+        }
+        this.listeners = Collections.unmodifiableList(new ArrayList<>(listeners));
+    }
+
+    @Override
+    public void onAfterTranslogSync() {
+        for (TranslogEventListener listener : listeners) {
+            try {
+                listener.onAfterTranslogSync();
+            } catch (Exception ex) {
+                logger.warn(() -> new ParameterizedMessage("failed to invoke onTranslogSync listener"), ex);
+                throw ex;
+            }
+        }
+    }
+
+    @Override
+    public void onAfterTranslogRecovery() {
+        for (TranslogEventListener listener : listeners) {
+            try {
+                listener.onAfterTranslogRecovery();
```

Comment:
@Bukhtawar sorry, I just noticed that, I believe we'll run into issue here by not calling all listeners (apply to all methods):

```
for (TranslogEventListener listener : listeners) {
    try {
        listener.onAfterTranslogRecovery();
    } catch (Exception ex) {
        logger.warn(() -> new ParameterizedMessage("failed to invoke onBeginTranslogRecovery listener"), ex);
    }
}
```
Even if one of the listeners fail, we should continue. Optionally, if it makes sense, the exception(s) could be collected and rethrown.

Reply:
Damn!! missed it. Thanks Nice catch